templates/php/src/Services/Service.php.twig

```diff
@@ -55,6 +55,44 @@ class {{ service.name | caseUcfirst }} extends Service
         }
 
 {% endfor %}
+{% if 'multipart/form-data' in method.consumes %}
+{% for parameter in method.parameters.all %}
+{% if parameter.type == 'file' %}
+        $size = filesize(${{ parameter.name | caseCamel }});
+        //send single file if size is less than or equal to 5MB
+        if ($size <= Client::CHUNK_SIZE) {
+            $params['{{ parameter.name }}'] = new \CURLFile(${{ parameter.name | caseCamel }});
+            return $this->client->call(Client::METHOD_{{ method.method | caseUpper }}, $path, [
+{% for param in method.parameters.header %}
+                '{{ param.name }}' => ${{ param.name | caseCamel }},
+{% endfor %}
+{% for key, header in method.headers %}
+                '{{ key }}' => '{{ header }}',
+{% endfor %}
+                ], $params);
+        } else {
+            $id = '';
+            $handle = @fopen(${{ parameter.name | caseCamel }}, "rb");
+            $mimeType = mime_content_type(${{ parameter.name | caseCamel }});
+            $counter = 0;
+            while (!feof($handle)) {
+                $params['{{ parameter.name }}'] = new \CURLFile('data://' . $mimeType . ';base64,' . base64_encode(@fread($handle, Client::CHUNK_SIZE)), $mimeType, basename(${{ parameter.name | caseCamel }}));
+                $contentRanges = 'bytes ' . ($counter * Client::CHUNK_SIZE) . '-' . min(((($counter * Client::CHUNK_SIZE) + Client::CHUNK_SIZE) - 1), $size) . '/' . $size;
+                $response = $this->client->call(Client::METHOD_POST, $path, [
+                    'content-type' => 'multipart/form-data',
+                    'content-range' => $contentRanges,
+                    'x-appwrite-id' => $id,
+                ], $params);
+                $counter++;
+                $id = $response['$id'];
+                $params['{{ method.parameters.all[1].name }}'] = $id; {# Asuming that with custom ID support, second parameter (fileID and executionID both are second param), but we don't have custom ID for tags yet so will fail for createTag #}
```

Comment:
@eldadfux Here, while making a chunked upload request, for subsequent request, we need the update the ID parameter (if we don't want to use extra header `x-appwrite-id`), but how will we know which parameter is the actual ID parameter?

Also, createTag, doesn't yet support custom ID so I think, using the header approach is best at the moment.